app/views/transactions/_transaction_group.html.erb

```diff
@@ -1,10 +1,13 @@
-<%# locals: (date:, transactions:) %>
-<div class="bg-gray-25 rounded-xl p-1">
+<%# locals: (transaction_group:, is_widget: "no") %>
```

Comment:
While I'm fine with the `is_widget` concept, I don't think we should be passing `is_widget` through so many partials (`_transaction_group`, `_menu`, `_row`) and storing in query params.  Many of these partials don't need to know about this config and are simply passing it along.

I think we could isolate this piece of logic better by encapsulating it directly in the `_transaction` partial and using a helper to determine the setting.  While this solution still doesn't feel like the _best_, it reduces the surface area for change and will allow us to update one file in the future if we do come up with a better solution (rather than having to pass locals to so many partials):

```rb
# transactions_helper.rb

def full_width_transaction_row?(route)
  route != "/"
end
```

```erb
<%# _transaction.html.erb %>

<%= debug full_width_transaction_row?(request.path) %> # If on dashboard, returns false, otherwise true
```

Reply:
I agree with you, I'm passing `is_widget` to many partials, and the code could be difficult to maintain. Unfortunately, I don't think that moving the logic inside a `transactions_helper.rb` will fix the problem:

1 - On the dashboard, `request.path` is "/";
2 - When I edit the transaction from the dashboard, `request.path` is "/transactions/:id";
3 - On the transactions page, `request.path` is "/transactions";
4 - When I edit the transaction from the transactions page, `request.path` is "/transactions/:id";

As you can see from points 2 and 4, `request.path` is the same (as we're pointing to `transactions#update`) but for point 2, we need the widget version, while for point 4, we need the full-width version. 

Probably, this could be a solution (there could be some mistakes, I just want to give an idea):

```ruby
# app/views/transactions/_transaction.html.erb

<%# locals: (transaction:, is_widget: "no") %>

<%= turbo_frame_tag dom_id(transaction), class:"text-gray-900 flex items-center gap-6 py-4 text-sm font-medium px-4" do %>
  <%= link_to transaction_path(transaction, is_widget:), data: { turbo_frame: "modal" }, class: "group", id: dom_id(transaction) do %>
    <%= render partial: "transactions/transaction_name" %>
  <% end %>
  <%= content_tag :div, class: "w-#{is_widget == 'yes' ? 36 : 48}" do %>
    <%= render partial: "transactions/categories/menu" %>
  <% end %>
  <%= render partial: "transactions/transaction_account_name" if is_widget == "no" %>
  <%= render partial: "transactions/transaction_amount" %>
<% end %>
``` 

```ruby
# app/controllers/transactions_controller.rb

def update
  respond_to do |format|
    if @transaction.update(transaction_params)
      @transaction.account.sync_later

      format.html { redirect_to transaction_url(@transaction), notice: t(".success") }
      format.turbo_stream do
        render turbo_stream: [
          turbo_stream.append("notification-tray", partial: "shared/notification", locals: { type: "success", content: { body: t(".success") } }),
          turbo_stream.replace("transaction_name_#{@transaction.id}", partial: "transactions/transaction_name", locals: { transaction: @transaction }),
          turbo_stream.replace("transaction_category_#{@transaction.id}", partial: "transactions/categories/menu", locals: { transaction: @transaction }),
          turbo_stream.replace("transaction_account_#{@transaction.id}", partial: "transactions/transaction_account_name", locals: { transaction: @transaction }),
          turbo_stream.replace("transaction_amount_#{@transaction.id}", partial: "transactions/transaction_amount", locals: { transaction: @transaction }),
        ]
      end
    else
      format.html { render :edit, status: :unprocessable_entity }
    end
  end
end
``` 

By doing so, I can leave `is_widget` only on the `_transaction` partial. The controller will render all, and when we are on the dashboard, Turbo will skip the `transactions/transaction_account_name` partial because it won't find the corresponding id.
What do you think about that?

Reply:
@mattia-malnis got it, I understand what you're saying now.  I was not considering the stream response that's being  constructed after editing.

While long term I'd imagine we'll introduce editing from the dashboard, it sounds like if we made it "read only" on the dashboard we could use the helper and avoid this.  Am I thinking about that correctly?

If so, I think we'd be totally fine to make this read only for now to avoid all the extra complexity.

Reply:
@zachgoll I updated the code following your advice on using the `full_width_transaction_row?` helper and making the widget read-only!

Reply:
Looks good!